// Copyright 2016 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto3";

option optimize_for = LITE_RUNTIME;

package blimp.helium;

message VersionVectorMessage {
  uint64 local_revision = 1;
  uint64 remote_revision = 2;
}

message ValueMessage {
  oneof value {
    int32 int32_value = 1;
    string string_value = 2;
  }
}

message LwwRegisterChangesetMessage {
  VersionVectorMessage last_modified = 1;
  ValueMessage value = 2;
}

message InputMethodChangesetMessage {
  LwwRegisterChangesetMessage input_generation = 1;
  LwwRegisterChangesetMessage input_type = 2;
  LwwRegisterChangesetMessage response_generation = 3;
  LwwRegisterChangesetMessage response = 4;
}
